### DISCUSSION TITLE
differences of libc.a under differenct directories

### DISCUSSION THREAD

**Discussion by dajiangzhong** [xpack-dev-tools/riscv-none-elf-gcc-xpack#31] [General]
Hi There, 
I am using xpack-riscv-none-embed-gcc-10.2.0-1.2 RSIC-V tool chain in my embedded project.  When I compile the source code of my project, I set my processor architecture as -march=rv32imc but didn't set -mabi's value.  The building process worked successfully. But I noticed there are multiple libc.a libraries scattered under the xpack-riscv-none-embed-gcc-10.2.0-1.2 directory. For my specific process, I found that I could use libc.a under xpack-riscv-none-embed-gcc-10.2.0-1.2 /lib or xpack-riscv-none-embed-gcc-10.2.0-1.2/riscv-none-embed/lib/rv32imc/ilp32 to link with my code successfully. But I check the two libc.a has different sha256 checksum respectively, so they are different libraries.  I am not clear, in the linking process which libc.a were used by gcc to produce the final program. 
Very appreciated if any one can help on clarify differences between the two libc.a libraries and which libc.a library were used in my final program.

**Comment by dajiangzhong** [comment]
Does the order of directories in environment variable LIBRARY_PATH determine the libc.a be used?

**Reply by ilg-ul** [reply]
This week I'll be out, with limited connectivity.

GCC 10 is a bit old, and the build scripts might have failed occasionally, so I don't know exactly, but normally the environment should not matter.
